assets/javascripts/discourse/initializers/add-search-results.js

```diff
@@ -0,0 +1,130 @@
+import { Promise } from "rsvp";
+import { ajax } from "discourse/lib/ajax";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import Post from "discourse/models/post";
+import Topic from "discourse/models/topic";
+import {
+  ENCRYPT_ACTIVE,
+  getEncryptionStatus,
+  getIdentity,
+} from "discourse/plugins/discourse-encrypt/lib/discourse";
+import {
+  decrypt,
+  importKey,
+} from "discourse/plugins/discourse-encrypt/lib/protocol";
+import I18n from "I18n";
+
+const CACHE_KEY = "discourse-encrypt-cache";
+
+function getCache() {
+  const cache = window.sessionStorage.getItem(CACHE_KEY);
```

Comment:
I am curious why we don't use the discourse session here? I believe it has the same lifetime (page) and is usually accessible via `this.session` or looking it up on the store in an initializer.